File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.cpp
class Solution {
public:
    double findMaxAverage(vector<int>& nums, int k) {
        double total = 0;
        for(int i = 0; i < k; i++) {
            total += nums[i];
        }
        double res = total;
        for(int i = k; i < nums.size(); i++) {
            total += nums[i] - nums[i-k];
            res = max(res, total);
        }
        return res/k;
    }
};

File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        total = 0
        for i in range(k):
            total += nums[i]
        res = total
        for i in range(k, len(nums)):
            total += nums[i] - nums[i-k]
            res = max(res,total)
        return res/k